crates/control/src/search_suggestor.rs

```diff
@@ -0,0 +1,149 @@
+use std::ops::{Index, IndexMut};
+
+use color_eyre::Result;
+use context_attribute::context;
+use coordinate_systems::{Field, Ground};
+use framework::{AdditionalOutput, MainOutput};
+use linear_algebra::{point, Isometry2, Point2};
+use nalgebra::{clamp, DMatrix};
+use serde::{Deserialize, Serialize};
+use types::{
+    ball_position::{BallPosition, HypotheticalBallPosition},
+    field_dimensions::FieldDimensions,
+    parameters::SearchSuggestorParameters,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct SearchSuggestor {
+    heatmap: Heatmap,
+}
+
+#[context]
+pub struct CreationContext {
+    field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    search_suggestor_configuration: Parameter<SearchSuggestorParameters, "search_suggestor">,
+}
+
+#[context]
+pub struct CycleContext {
+    search_suggestor_configuration: Parameter<SearchSuggestorParameters, "search_suggestor">,
+    ball_position: Input<Option<BallPosition<Ground>>, "ball_position?">,
+    invalid_ball_positions: Input<Vec<HypotheticalBallPosition<Ground>>, "invalid_ball_positions">,
```

Comment:
what about `removed_all_positions`

Reply:
These will be relevant for decreasing the validity for checked areas, which is a thing for the next iteration of ball search.